components/layout_2020/replaced.rs

```diff
@@ -563,59 +565,47 @@ impl ReplacedContent {
             )
             .into()
         });
+        let preferred_inline =
+            box_size
+                .inline
+                .resolve(Size::FitContent, inline_stretch_size, &inline_content_size);
+        let min_inline = min_box_size
+            .inline
+            .resolve_non_initial(inline_stretch_size, &inline_content_size)
+            .unwrap_or_default();
+        let max_inline = max_box_size
+            .inline
+            .resolve_non_initial(inline_stretch_size, &inline_content_size);
+        let inline_size = preferred_inline.clamp_between_extremums(min_inline, max_inline);
+
+        // Now we can compute the block size, using the inline size from above.
         let block_content_size = LazyCell::new(|| -> ContentSizes {
-            let get_inline_size = || {
-                SizeConstraint::new(
-                    box_size
-                        .inline
-                        .maybe_resolve_extrinsic(Some(inline_stretch_size)),
```

Comment:
The bug was here, this would be the minimal fix:

```diff
--- a/components/layout_2020/replaced.rs
+++ b/components/layout_2020/replaced.rs
@@ -568,7 +568,7 @@ impl ReplacedContent {
                 SizeConstraint::new(
                     box_size
                         .inline
-                        .maybe_resolve_extrinsic(Some(inline_stretch_size)),
+                        .resolve_non_initial(inline_stretch_size, &inline_content_size),
                     min_box_size
                         .inline
                         .resolve_non_initial(inline_stretch_size, &inline_content_size)
```

But I'm simplifying the logic while I'm at it.